### ISSUE TITLE
indenting errors with `class_exists` checks

### ISSUE THREAD

**Issue by norcross** [WordPress-Coding-Standards/WordPress-Coding-Standards#416]
when running the sniffer, I get indentation errors when my class setup is done like this:
```php
if ( ! class_exists( 'My_Cool_Class' ) ) {

/**
 * My Cool Class.
 *
 * Contains all the cool class stuff.
 */
class My_Cool_Class {
/// blah blah blah
```
it expects the entire class to be indented. All the examples I've seen using this check don't handle it that way, so I'm not sure if there's an exclusion to be added, or if I'm just stubborn.

**Comment by JDGrimes** [comment]
I usually see this used without indentation as well. As usual, WordPress core is inconsistent here. Most of the classes that are wrapped in `class_exists()` checks are part of external libraries, or are deprecated. Among these groups, most of the classes are not indented, and use `if ( /*...*/ ) : /* ... */ endif;` syntax. But there are exceptions to both of these rules.

I only found one place in core where a non-deprecated class was wrapped in a `class_exists()` check. This was for `WP_Image_Editor_Mock`, which is a part of the unit tests, and was actually a check for the existence of the parent class (`WP_Image_Editor`). FWIW, it was indented and used the colon syntax.

It would seem to me that WordPress core's standard doesn't define how classes should be indented in `class_exists()` checks, because it seems to practice a philosophy of not using them.

**Comment by norcross** [comment]
I follow. I've also checked some stuff outside of WP and most don't indent either. I'm guessing at this point I should just ignore it? Or do you know of a way to exclude that from the check?

**Comment by JDGrimes** [comment]
I think this check can be excluded using the `ignoreIndentationTokens` property of the `Generic.WhiteSpace.ScopeIndent` sniff (I'm assuming that that is the sniff causing the error). The property is an array of Token IDs. So if you want to configure it in your XML config file, you can't just put `T_CLASS`, you have to know the value of `T_CLASS` (which can change in different PHP versions, I think). The value is `358` in PHP 5.6.7. I do not know if it is different in other supported versions or not.

You would add something like this to your custom XML config file:

```xml
	<rule ref="Generic.WhiteSpace.ScopeIndent">
		<properties>
			<property name="ignoreIndentationTokens" value="358" type="array" />
		</properties>
	</rule>
```

Otherwise you could create a subclass of that sniff and define the property directly.

**Comment by norcross** [comment]
thanks, I'll see what I can get working. I appreciate the help (I'll close this for now to keep your queue clean)
